[Jian Rong] iP

kuanjr opened the pull request:
# DUKE

> "Clutter is nothing more than postponed deadlines" ~ Barbara Hemphill ([source](https://nacm.org/pdfs/webinars/clutter-is-postponed-exercise-1031.pdf))

Duke is an organisational tool that helps you stay clutter free and keep track of your tasks!
It is:
 - Text-based
 - Easy to use
 - ~~Helpful~~ Extremely helpful

You can:
1. Add 3 kinds of tasks - Todo, Dealines, Events
2. Mark the tasks as done
3. List them down for easy viewing
4. Delete items once you are done with them 👍

Examples of code syntax
To add Events, simply write `event <Name of Event> /at YYYY:MM:DD HH:MM`
> event Camp /at 2022:08:09 08:00


## List of Features
 - [x] Add 3 new tasks: Todo. Deadline, Event
 - [ ] Edit tasks
 - [ ] Export to calendar

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public static void main(String[] args) {
    new Duke().listen();
}
```

Commit:
Add Gradle support

kuanjr committed:
added Level-1

kuanjr committed:
added Level-1

kuanjr committed:
Merge branch 'master' of https://github.com/kuanjr/ip

kuanjr committed:
added Level-2

kuanjr committed:
added Level-3

kuanjr committed:
added Task class, Todo, Deadline and Event subclass

kuanjr committed:
Added Level-4

kuanjr committed:
text-ui-testing

kuanjr committed:
text-ui-testing

kuanjr committed:
text-ui-testing

kuanjr committed:
added Level-5

kuanjr committed:
added Level-5

kuanjr committed:
Merge branch 'master' of https://github.com/kuanjr/ip

kuanjr committed:
added Level-6

kuanjr committed:
Merge branch 'master' of https://github.com/kuanjr/ip

kuanjr committed:
added Level-6

kuanjr committed:
changed indentation of switch-case

kuanjr committed:
Add write to file function

kuanjr committed:
Add Date and Time format for Deadline and Event

kuanjr committed:
Merge branch 'branch-Level-8'

kuanjr committed:
Abstract basic functionality of Ui and Parser to seperate classes

kuanjr committed:
Abstract the functionalities of Ui, Storage, Parser and TaskList into seperate classes

shurvirarora commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,13 @@
+public abstract class Task {
+
+    boolean checked = false;
```
boolean should be named "isChecked" instead, according to Java Coding Standard 👍

shurvirarora commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,13 @@
+        this.checked = true;
+    }
+
+    public void setUnChecked(){
```
perhaps the naming for this could be "setUnchecked()" instead 👍

shurvirarora commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,20 @@
+        System.out.println("added: " + this.toString());
+    }
+
+    public String toString(){
```
There is some inconsistency with the white spacing (for e.g. there should be a white space between "toString()" and '{' according to Java Coding Standard

shurvirarora reviewed (commented):
LGTM. Just some nits to fix (e.g white spaces, boolean naming)

kuanjr committed:
Divide class into package.

jbkim1999 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,27 @@
+
+    private void listen(){
+        Ui.printLogo();
+        boolean running = true;
```
Have you considered renaming your boolean variable to isRunning?

jbkim1999 commented on src/main/java/duke/command/Storage.java:
```diff
@@ -0,0 +1,34 @@
+        this.path = path;
+        this.taskList = taskList;
+        try {
+            File myObj = new File(path);
```
What about myFile as a name for File instead? I feel like object might be too general!

jbkim1999 commented on src/main/java/duke/command/Ui.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    public static boolean run(TaskList taskList) {
+        Scanner dukeScan = new Scanner(System.in);
```
How about dukeScanner instead of dukeScan (which may be interpreted as a verb as well)?

jbkim1999 reviewed (commented):
Overall, I think you are naming your variables very clearly! I suggested a few changes, feel free to change them if you like my suggestions! Thank you :)

alexteo98 commented on src/main/java/duke/command/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+            System.out.println("No items in the list");
+        } else {
+            for (int i = 0; i < taskArray.size(); i++) {
+                System.out.printf("%d. " + taskArray.get(i)+ "\n", i+1);
```
There should be whitespaces before the '+' operators.

alexteo98 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,47 @@
+        String[] tempArray = Arrays.copyOfRange(inputArray, 1, inputArray.length);
+        String [] command = {inputArray[0], String.join(" ", tempArray)};
+        try {
+            switch (command[0]) {
+            case ("bye"):
+              return false;
+            case ("list"):
+                taskList.listItem();
+                break;
+            case ("mark"):
+                taskList.markItem(command);
+                break;
+            case ("unmark"):
+                taskList.unmarkItem(command);
+                break;
+            case ("delete"):
+                taskList.deleteItem(command);
+                break;
+            case ("todo"):
+            case ("deadline"):
+            case ("event"):
+                taskList.addItem(command);
+                break;
+            default:
+                throw new DukeException("Sorry I don't understand that command");
+            }
```
Good job on the correct indentation of this switch block!

alexteo98 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+            String[] deadlineList = deadline.split(" ");
+            this.date = LocalDate.parse(deadlineList[0].replace("/", "-"));
+            this.time = LocalTime.parse(deadlineList[1]);
+        } catch (IndexOutOfBoundsException | DateTimeParseException e){
```
There are supposed to be whitespaces before opening the curly braces.

alexteo98 reviewed (commented):
Generally well done on following coding standards, some errors are present, but CheckStyle can find most of them. Good job!

kuanjr committed:
Add JUnit tests to test the behavior of the code

kuanjr committed:
Add JUnit tests to test the behavior of the code.

kuanjr committed:
Change implementation of Storage

kuanjr committed:
Add javadocs to classes and methods.

kuanjr committed:
Tweak code to match coding standard.

kuanjr committed:
Tweak code to match coding standard.

kuanjr committed:
Add search feature

kuanjr committed:
fix merge conflicts between master and branch-A-CodingStandard

kuanjr committed:
Fix conflicts between master and branch-Level-9

kuanjr committed:
Fix conflict between remote and master

kuanjr committed:
Merge branch 'add-gradle-support'

kuanjr committed:
Add Gradle support

kuanjr committed:
Implement GUI with JavaFX and FXML

kuanjr committed:
Integrate Duke functionality into GUI

kuanjr committed:
Add Assertions

kuanjr committed:
Merge pull request #2 from kuanjr/branch-A-Assertions

Add Assertions

kuanjr committed:
Improve code quality

kuanjr committed:
Merge pull request #3 from kuanjr/branch-A-CodeQuality

Improve code quality

kuanjr committed:
Add priority variable to tasks.

There are 3 priorities: low, medium, high

When a task is created, the priority is automatically set to low.

Users can set priority of a task by using this command:
"priority <index> <priority>"
Eg. priority 1 high

kuanjr committed:
Fix Find feature

Find feature was previously left out during integration with GUI. It is
now fixed.

kuanjr committed:
Add User Guide and screenshot of product

User Guide is done in GFMD and includes details on Duke's features,
command summary and instructions on how to start.

kuanjr committed:
Fix links in User Guide

kuanjr committed:
Add link to Duke Jar file
